homeassistant/components/subaru/sensor.py

```diff
@@ -0,0 +1,251 @@
+
+from .const import (
+    API_GEN_2,
+    DOMAIN,
+    ENTRY_COORDINATOR,
+    ENTRY_VEHICLES,
+    VEHICLE_API_GEN,
+    VEHICLE_HAS_EV,
+    VEHICLE_HAS_SAFETY_SERVICE,
+)
+from .entity import SubaruEntity
+
+L_PER_GAL = vol_convert(1, VOLUME_GALLONS, VOLUME_LITERS)
+KM_PER_MI = dist_convert(1, LENGTH_MILES, LENGTH_KILOMETERS)
+
+# Fuel Economy Constants
+FUEL_CONSUMPTION_L_PER_100KM = "L/100km"
+FUEL_CONSUMPTION_MPG = "mi/gal"
+FUEL_CONSUMPTION_UNITS = [FUEL_CONSUMPTION_L_PER_100KM, FUEL_CONSUMPTION_MPG]
+
+SENSOR_TYPE = "type"
+SENSOR_FIELD = "field"
+SENSOR_UNITS = "units"
+
+# Sensor data available to "Subaru Safety Plus" subscribers with Gen1 or Gen2 vehicles
+SAFETY_SENSORS = [
+    {
+        SENSOR_TYPE: "Odometer",
+        SENSOR_FIELD: sc.ODOMETER,
+        SENSOR_UNITS: LENGTH_KILOMETERS,
+    },
+]
+
+# Sensor data available to "Subaru Safety Plus" subscribers with Gen2 vehicles
+API_GEN_2_SENSORS = [
+    {
+        SENSOR_TYPE: "Avg Fuel Consumption",
+        SENSOR_FIELD: sc.AVG_FUEL_CONSUMPTION,
+        SENSOR_UNITS: FUEL_CONSUMPTION_L_PER_100KM,
+    },
+    {
+        SENSOR_TYPE: "Range",
+        SENSOR_FIELD: sc.DIST_TO_EMPTY,
+        SENSOR_UNITS: LENGTH_KILOMETERS,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure FL",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_FL,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure FR",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_FR,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure RL",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_RL,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "Tire Pressure RR",
+        SENSOR_FIELD: sc.TIRE_PRESSURE_RR,
+        SENSOR_UNITS: PRESSURE_HPA,
+    },
+    {
+        SENSOR_TYPE: "External Temp",
+        SENSOR_FIELD: sc.EXTERNAL_TEMP,
+        SENSOR_UNITS: TEMP_CELSIUS,
+    },
+    {
+        SENSOR_TYPE: "12V Battery Voltage",
+        SENSOR_FIELD: sc.BATTERY_VOLTAGE,
+        SENSOR_UNITS: VOLT,
+    },
+]
+
+# Sensor data available to "Subaru Safety Plus" subscribers with PHEV vehicles
+EV_SENSORS = [
+    {
+        SENSOR_TYPE: "EV Range",
+        SENSOR_FIELD: sc.EV_DISTANCE_TO_EMPTY,
+        SENSOR_UNITS: LENGTH_MILES,
+    },
+    {
+        SENSOR_TYPE: "EV Battery Level",
+        SENSOR_FIELD: sc.EV_STATE_OF_CHARGE_PERCENT,
+        SENSOR_UNITS: PERCENTAGE,
+    },
+    {
+        SENSOR_TYPE: "EV Time to Full Charge",
+        SENSOR_FIELD: sc.EV_TIME_TO_FULLY_CHARGED,
+        SENSOR_UNITS: TIME_MINUTES,
+    },
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Subaru sensors by config_entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][ENTRY_COORDINATOR]
+    vehicle_info = hass.data[DOMAIN][config_entry.entry_id][ENTRY_VEHICLES]
+    entities = []
+    for vin in vehicle_info.keys():
+        _create_sensor_entities(entities, vehicle_info[vin], coordinator)
```

Comment:
Return the entities and append them to the list instead of mutating entities in the function